pkg/storage/postgresql.go

```diff
@@ -5,14 +5,21 @@ import (
 	"log/slog"
 	"time"
 
-	"github.com/dhis2-sre/im-manager/pkg/config"
 	"github.com/dhis2-sre/im-manager/pkg/model"
 	slogGorm "github.com/orandin/slog-gorm"
 	"gorm.io/driver/postgres"
 	"gorm.io/gorm"
 )
 
-func NewDatabase(logger *slog.Logger, c config.Postgresql) (*gorm.DB, error) {
+type PostgresqlConfig struct {
+	Host         string
+	Port         int
+	Username     string
+	Password     string
+	DatabaseName string
+}
+
+func NewDatabase(logger *slog.Logger, c PostgresqlConfig) (*gorm.DB, error) {
```

Comment:
Not this line, but on line 34-38 we're assigning each property from the struct to variables of the same name. How about skipping that and just used the struct?